### PR TITLE
Overlap data creation and ir_value setting with previous execution

### PR DESCRIPTION
**Pull request by aws-rhsoln**
Upstreaming changes from torch_xla to lazy tensor core: https://github.com/pytorch/xla/pull/4011

Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/csrc/lazy/core/lazy_graph_executor.cpp [hunk 1]
```diff
@@ -679,13 +679,37 @@ std::vector<Value> LazyGraphExecutor::CollectRoots(
   return roots;
 }
 
-std::vector<BackendDataPtr> LazyGraphExecutor::FetchTensorData(
+void LazyGraphExecutor::ExtractIRAndPrepareXlaData(
```
Comment (alanwaketan) [inline torch/csrc/lazy/core/lazy_graph_executor.cpp:682]: Can you omit `Xla` in the name? Change it to `ExtractIRAndPrepareTensorData`

File: torch/csrc/lazy/core/lazy_graph_executor.h [hunk 2]
```diff
@@ -192,24 +192,35 @@ class TORCH_API LazyGraphExecutor {
...
 
-  std::vector<BackendDataPtr> FetchTensorData(
+  std::vector<BackendDataPtr> SetTensorData(
       std::vector<LazyTensorPtr>* tensors,
       const SyncTensorsConfig& config,
-      c10::ArrayRef<size_t> indices);
+      c10::ArrayRef<size_t> indices,
+      const std::vector<torch::lazy::BackendDataPtr>& tensor_data_vec);
+
+  void ExtractIRAndPrepareXlaData(
```
Comment (alanwaketan) [inline torch/csrc/lazy/core/lazy_graph_executor.h:201]: Same here, replace `Xla` with `Tensor`.

**Review by alanwaketan — APPROVED** [review]
Overall, it looks matching the XLA pull request: pytorch/xla#4011.

File: torch/csrc/lazy/core/lazy_graph_executor.cpp (hunk 1 above)
Comment (JackCaoG) [reply torch/csrc/lazy/core/lazy_graph_executor.cpp:682]: bump

Comment (aws-rhsoln) [reply torch/csrc/lazy/core/lazy_graph_executor.cpp:682]: yeah, let me make the change

**Comment by alanwaketan** [comment]
@aws-rhsoln Just wonder if we are ready to land this change?

**Comment by aws-rhsoln** [comment]
> @aws-rhsoln Just wonder if we are ready to land this change?

Sorry, I missed this. Yeah we were ready some time back, but I think now I see some merge conflict. Let me resolve the conflict and send a new revision.

**Comment by alanwaketan** [comment]
Hi @aws-rhsoln, the PyTorch/XLA team is actively working on LTC migration and this PR is crucial for us to unify the LazyGraphExecutor and XLAGraphExecutor. Therefore, I'm wondering if you can land this soon. If you don't have the bandwidth, would you mind if I take over this PR?

**Comment by alanwaketan** [comment]
Per discussion over Slack, I'm taking over this PR. Given it was developed on a fork, I will need to close this PR and clone a new one.

**Pull request closed without merging**